Src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/NativeViewExpansion.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Debugger;
+using Microsoft.VisualStudio.Debugger.Evaluation;
+using Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation;
+using System;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal sealed class NativeViewExpansion : Expansion
+    {
+        internal override void GetRows(
+            ResultProvider resultProvider,
+            ArrayBuilder<EvalResultDataItem> rows,
+            DkmInspectionContext inspectionContext,
+            EvalResultDataItem parent,
+            DkmClrValue value,
+            int startIndex,
+            int count,
+            bool visitAll,
+            ref int index)
+        {
+            if (InRange(startIndex, count, index))
+            {
+                rows.Add(GetRow(resultProvider, inspectionContext, value, parent));
+            }
+
+            index++;
+        }
+
+        private EvalResultDataItem GetRow(
+            ResultProvider resultProvider,
+            DkmInspectionContext inspectionContext,
+            DkmClrValue comObject,
+            EvalResultDataItem parent)
+        {
+            try
```

Comment:
Can this be determined when we create the Expansion (so that we don't have to throw/catch an Exception for every row?)